nexus/src/external_api/console_api.rs

```diff
@@ -89,6 +91,103 @@ pub async fn spoof_login(
         .body("ok".into())?) // TODO: what do we return from login?
 }
 
+#[derive(Deserialize, JsonSchema)]
+pub struct LoginToProviderPathParam {
+    pub silo_name: crate::db::model::Name,
+    pub provider_name: crate::db::model::Name,
+}
+
+/// Ask the user to login to their identity provider
+///
+/// Either display a page asking a user for their credentials, or redirect them
+/// to their identity provider.
+#[endpoint {
+   method = GET,
+   path = "/login/{silo_name}/{provider_name}",
+   tags = ["login"],
+}]
+pub async fn ask_user_to_login_to_provider(
```

Comment:
tbh I find the level of detail in this name more confusing than helpful. This is supposed to fully replace my fake [`/login`](https://github.com/oxidecomputer/omicron/blob/3ae28c77cbd2355abeab07e358c5ccbeb035f641/nexus/src/external_api/console_api.rs#L192-L208) right? I think `login` would be a fine name for this. It wouldn't matter at all, but this name ends up in the generated clients as the name of the method and its params type, and it threw me off a bit:

```ts
/**
 * Ask the user to login to their identity provider
 */
askUserToLoginToProvider: (
  { providerName, siloName }: AskUserToLoginToProviderParams,
  params: RequestParams = {}
) =>
  this.request<void>({
    path: `/login/${siloName}/${providerName}`,
    method: 'GET',
    ...params,
  }),
```

In my experience it's common to see a `GET /login(/whatever)` and `POST /login` in an API and they do exactly what we're doing here.

Reply:
renamed in 20fdc111